core/gui/src/app/workspace/service/report-generation/report-generation.service.ts

```diff
@@ -67,27 +67,231 @@ export class ReportGenerationService {
   }
 
   /**
-   * Generates an HTML file containing the workflow snapshot and triggers a download of the file.
-   * @param {string} workflowSnapshotURL - The base64-encoded PNG image URL of the workflow snapshot.
-   * @param {string} workflowName - The name of the workflow.
+   * Collects all operator results from the workflow and generates a comprehensive HTML report.
+   * @param {Object} payload - The payload containing operator IDs, operator details, workflow snapshot, and workflow name.
+   * @param {string[]} payload.operatorIds - Array of operator IDs present in the workflow.
+   * @param {any[]} payload.operators - Array of operator details, including configurations and metadata.
+   * @param {string} payload.workflowSnapshot - Base64-encoded PNG image URL of the workflow snapshot.
+   * @param {string} payload.workflowName - The name of the workflow, used for naming the final report.
+   * @returns {void}
+   */
+
+  public getAllOperatorResults(payload: {
+    operatorIds: string[];
+    operators: any[];
+    workflowSnapshot: string;
+    workflowName: string;
+  }): void {
+    console.log("Starting getAllOperatorResults with payload: ", payload);
+    console.log("Type of payload.operatorIds: ", typeof payload.operatorIds);
+    console.log("Is Array: ", Array.isArray(payload.operatorIds));
+
+    const allResults: { operatorId: string; html: string }[] = [];
+
+    const promises = payload.operatorIds.map(operatorId => {
+      const operatorInfo = payload.operators.find((op: any) => op.operatorID === operatorId);
+      return this.displayResult(operatorId, operatorInfo, allResults);
+    });
+
+    Promise.all(promises)
+      .then(() => {
+        console.log("All results before sorting: ", allResults);
+        console.log("Payload: ", payload);
+
+        const sortedResults = payload.operatorIds.map(
+          id => allResults.find(result => result.operatorId === id)?.html || ""
+        );
+
+        console.log("All results after sorting: ", sortedResults);
+
+        // Pass the workflowName to generateReportAsHtml
+        this.generateReportAsHtml(payload.workflowSnapshot, sortedResults, payload.workflowName);
+        console.log("Completed getAllOperatorResults");
+      })
+      .catch(error => {
+        console.error("Error in getAllOperatorResults: ", error);
```

Comment:
use notification service to log this message, similarly when report starts to generate, use the notification service